fulltext/__init__.py

```diff
@@ -383,11 +386,26 @@ def get(path_or_file, default=SENTINAL, mime=None, name=None, backend=None,
      * `mime` and `name` should be passed if the information
        is available to caller, otherwise a best guess is made.
        If both are specified `mime` takes precedence.
+     * `encoding` and `encoding_errors` are used to handle text encoding.
+       They are taken into consideration mostly only by pure-python
+       backends which do not rely on CLI tools.
+       Default to "utf8" and "strict" respectively.
      * `kwargs` are passed to the underlying backend.
     """
+    if encoding is None:
+        encoding = ENCODING
+    if encoding_errors is None:
+        encoding_errors = ENCODING_ERRORS
+
+    kwargs = kwargs or {}
+    kwargs.setdefault("mime", mime)
+    kwargs.setdefault("encoding", encoding)
+    kwargs.setdefault("encoding_errors", encoding_errors)
```

Comment:
Would you rather just do: `kwargs['encoding'] = encoding` here, otherwise, caller could potentially pass both, since encoding is only used in the backend `__init__.py` does not actually use the value, only defines it, there is no reason for caller to pass it in `kwargs` doing so seems to be invalid/error.

IOW, if the user defines both, the main args will be ignored.

Reply:
Not sure I understand. `kwargs['encoding'] = encoding` would still pass the arg to the underlying backend function.
Please note that with this PR all backend functions will receive `encoding` and `encoding_errors` args. Is this what you're not OK with?

Reply:
Just this... As a user of `fulltext`, if I do:

```python
fulltext.get(..., encoding='utf8', encoding_errors='strict',
             kwargs={'encoding': 'latin1', 'encoding_errors': 'ignore'})
```

I have sent conflicting arguments, because of `setdefault`, the backend will receive `latin1/ignore`, meaning those arguments currently take precedence. I just think that the args to `get()` should overwrite those values (take precedence).

Reply:
I understand and agree.

Reply:
For that matter, `mime` is kinda the same deal, but I can almost buy that the user may wish to use a different value for mime in `get()` vs. the backend. Not sure though. Just seems ambiguous, but probably not worth too much time or discussion. I just wanted to mention it.

Also, I just noticed that this code (which I believe I wrote) will modify kwargs in-situ. It should probably copy it before mutating the dictionary.

Reply:
You're right, the dict should be copied. Will do now.